### PR TITLE
CodeQL: Speed up Maven build

### PR DESCRIPTION
**Pull request by kohlschuetter**
For CodeQL anaylsis, we can ignore Checkstyle and License goals. Tthese goals are covered the CI phase that runs concurrently.

Local testing indicates a significant build time improvement (2:33 min -> 1:36 min).

Given that the maven phase in GitHub's CodeQL takes around 20 minutes, we may save quite some time.

Signed-off-by: Christian Kohlschütter <christian@kohlschutter.com>

### CODE REVIEW

**Comment by olamy** [comment]
There is a profile called fast for this ;)

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -80,7 +80,8 @@ jobs:
           mvn -e -B -V org.apache.maven.plugins:maven-wrapper-plugin:3.1.0:wrapper "-Dmaven=3.8.6"
 
       - name: Clean install dependencies and build
-        run: mvn clean install -DskipTests -B -V
+        run: mvn clean install -DskipTests -B -V -Dcheckstyle.skip=true -Dlicense.skip=true -Dmaven.site.skip=true -Dmaven.javadoc.skip=true
```
Comment (joakime) [inline .github/workflows/codeql-analysis.yml]: How about this way instead ...

```suggestion
        run: mvn clean install -DskipTests -B -V -Pfast
```

**Review by joakime — DISMISSED** [review]
Try the `fast` profile first.

File: .github/workflows/codeql-analysis.yml [hunk 2]
```diff
@@ -80,7 +80,7 @@ jobs:
           mvn -e -B -V org.apache.maven.plugins:maven-wrapper-plugin:3.1.0:wrapper "-Dmaven=3.8.6"
 
       - name: Clean install dependencies and build
-        run: mvn clean install -DskipTests -B -V -Pfast -Dmaven.javadoc.skip=true -T4
+        run: mvn clean install -DskipTests -B -V -Pfast -Dmaven.javadoc.skip=true -Dorg.slf4j.simpleLogger.dateTimeFormat=HH:mm:ss,SSS -Dorg.slf4j.simpleLogger.showDateTime=true
```
Comment (joakime) [inline .github/workflows/codeql-analysis.yml]: We should add javadoc.skip to the `fast` profile.

Comment (kohlschuetter) [reply .github/workflows/codeql-analysis.yml]: Good idea. However, I'm not seeing any improvements on Github's end (still over 17 minutes for that command). Currently debugging to see what's going on

**Comment by kohlschuetter** [comment]
Abandoning the effort to speed up CodeQL. The CodeQL action is slow because their VM is heavily throttled, not much we can do here.

**Pull request closed without merging**